### PR TITLE
BNS domains that resolve to addresses now searchable

### PR DESCRIPTION
**Pull request by stjet**
tested and working locally

### CODE REVIEW

File: src/app/services/api/api.service.ts [hunk 1]
```diff
@@ -204,6 +204,17 @@ export class ApiService {
             .toPromise();
     }
 
+    /** Fetch/query BNS domain. */
+    async fetchBNSDomain(domain_name: string, tld: string): Promise<any> {
+        await this._hasPingedApi();
+        return this._http
+            .post<any>(`${this.httpApi}/v1/account/bns`, {
```
Comment (dev-ptera) [inline src/app/services/api/api.service.ts:211]: Could we get some stronger typings here?  The api is versioned so we should know what schema to expect.

File: src/app/pages/account/account.component.ts [hunk 2]
```diff
@@ -111,13 +111,28 @@ export class AccountComponent implements OnDestroy {
...
+            const parts = address.split('.');
+            if (parts.length === 2) {
+                //search in api
+                try {
+                    const domain = await this.apiService.fetchBNSDomain(parts[0], parts[1]);
+                    if (domain.domain?.resolved_address) {
+                        return this._redirectToAccountPage(domain.domain?.resolved_address);
+                    }
+                } catch (_) {}
+            }
```
Comment (dev-ptera) [inline src/app/pages/account/account.component.ts:135]: Could we get a quick comment here explaining what this does?  If not a ban address, assume it's BNS?

File: src/app/services/search/search.service.ts [hunk 3]
```diff
@@ -47,4 +47,10 @@ export class SearchService {
     isValidBlock(block: string): boolean {
         return block && block.length === 64;
     }
+
+    isValidBNSDomain(bns: string): boolean {
+        const parts = bns.split('.');
+        //later, can also check for illegal characters once that is more settled
+        return parts.length === 2 && parts[0].length <= 32;
+    }
```
Comment (dev-ptera) [inline src/app/services/search/search.service.ts:55]: Could this be placed in the UtilService & then used here & in the AccountComponent?

File: src/app/pages/account/account.component.ts [hunk 4]
```diff
@@ -111,13 +111,28 @@ export class AccountComponent implements OnDestroy {
...
     /** Given a ban address, searches for account. */
-    private _searchAccount(address): void {
+    private async _searchAccount(address): Promise<void> {
         if (!address) {
             return;
         }
 
         if (!address.startsWith('ban_')) {
+            const parts = address.split('.');
+            if (parts.length === 2) {
```
Comment (dev-ptera) [inline src/app/pages/account/account.component.ts:127]: Use `isValidBNSDomain` utility function here?

**Comment by stjet** [comment]
@dev-ptera Hopefully that should address the review. I did add `banani-bns` as a dependency for the `Domain` type but maybe it only needs `@types/banani-bns`? Not quite sure how that works.

**Pull request merged**